compiler/src/iree/compiler/Codegen/Dialect/PCF/Transforms/test/convert_sref_to_memref.mlir

```diff
@@ -0,0 +1,464 @@
+      scf.yield %b#0, %b#1 : !pcf.sref<?x?xi32, #pcf.test_scope>, !pcf.sref<?x?xi32, #pcf.test_scope>
+    }
+    pcf.return
+  }
+  util.return
+}
+
+// CHECK-LABEL: @convert_for_loop_swap_conflict
+//  CHECK-SAME:     %[[ARG0:[A-Za-z0-9_]+]]: memref<?x?xi32, strided<[3, 1]>, 3>
+//  CHECK-SAME:     %[[ARG1:[A-Za-z0-9_]+]]: memref<?x?xi32, strided<[4, 1]>, 3>
+//       CHECK:   pcf.generic scope(#pcf.test_scope)
+//  CHECK-NEXT:     execute[{{.*}}] {
+//   CHECK-DAG:     %[[CAST0:.+]] = memref.cast %[[ARG0]] : memref<?x?xi32, strided<[3, 1]>, 3> to memref<?x?xi32, strided<[?, 1]>, 3>
+//   CHECK-DAG:     %[[CAST1:.+]] = memref.cast %[[ARG1]] : memref<?x?xi32, strided<[4, 1]>, 3> to memref<?x?xi32, strided<[?, 1]>, 3>
+//  CHECK-NEXT:     scf.for {{.*}} iter_args(%[[ITER0:.+]] = %[[CAST0]], %[[ITER1:.+]] = %[[CAST1]])
+//  CHECK-NEXT:       %[[B:.+]]:2 = util.optimization_barrier %[[ITER1]], %[[ITER0]]
+//  CHECK-NEXT:       scf.yield %[[B]]#0, %[[B]]#1
+//       CHECK:     pcf.return
+
+// -----
+
+func.func @convert_alloc(%d0: index) -> !pcf.sref<?x5xi32, #pcf.sequential> {
+  %0 = pcf.alloc(%d0) : !pcf.sref<?x5xi32, #pcf.sequential>
+  return %0 : !pcf.sref<?x5xi32, #pcf.sequential>
+}
+
+// CHECK-LABEL: @convert_alloc
+//  CHECK-SAME:   %[[D0:[A-Za-z0-9]+]]: index
+//       CHECK:   %[[ALLOC:.+]] = memref.alloc(%[[D0]]) {alignment = 16 : i64} : memref<?x5xi32>
+//       CHECK:   return %[[ALLOC]] : memref<?x5xi32>
+
+// -----
+
+util.func private @convert_get_memref(%arg0: memref<?x?xi32, strided<[?, 1]>, 3>, %s0: index, %s1: index) {
+  pcf.generic scope(#pcf.test_scope)
+    execute(%ref = %arg0)[%id: index, %n: index]
+         : (!pcf.sref<?x?xi32, #pcf.test_scope>)
+        -> (memref<?x?xi32, strided<[?, 1]>, 3>) {
+    %view = pcf.get_memref %ref[0, 1] [%s0, %s1] [1, 1] : !pcf.sref<?x?xi32, #pcf.test_scope> to memref<?x?xi32, strided<[?, ?], offset: ?>>
+    util.optimization_barrier %view : memref<?x?xi32, strided<[?, ?], offset: ?>>
+    pcf.return
+  }
+  util.return
+}
+
+// CHECK-LABEL: @convert_get_memref
+//  CHECK-SAME:     %[[ARG0:[A-Za-z0-9_]+]]: memref<?x?xi32, strided<[?, 1]>, 3>
+//  CHECK-SAME:     %[[S0:[A-Za-z0-9_]+]]: index
+//  CHECK-SAME:     %[[S1:[A-Za-z0-9_]+]]: index
+//       CHECK:   pcf.generic
+//  CHECK-NEXT:     execute[{{.*}}] {
+//   CHECK-DAG:     %[[CAST:.+]] = memref.memory_space_cast %[[ARG0]] : memref<?x?xi32, strided<[?, 1]>, 3> to memref<?x?xi32, strided<[?, 1]>>
+//   CHECK-DAG:     %[[CAST2:.+]] = memref.cast %[[CAST]] : memref<?x?xi32, strided<[?, 1]>> to memref<?x?xi32, strided<[?, ?], offset: ?>>
+//   CHECK-DAG:     %[[SV:.+]] = memref.subview %[[CAST2]][0, 1] [%[[S0]], %[[S1]]] [1, 1] : memref<?x?xi32, strided<[?, ?], offset: ?>> to memref<?x?xi32, strided<[?, ?], offset: ?>>
+//       CHECK:     util.optimization_barrier %[[SV]]
+//       CHECK:     pcf.return
+
+// -----
+
+util.func private @convert_get_memref_no_space(%arg0: memref<?x?xi32, strided<[?, 1]>>, %s0: index, %s1: index) {
```

Comment:
nit: Maybe put it above the previous test case since this is the less specialized case?